src/test/scala/is/hail/utils/BinaryHeapSuite.scala

```diff
@@ -0,0 +1,195 @@
+package is.hail.utils
+
+import org.scalatest._
+import Matchers._
+import org.testng.annotations.Test
+
+class BinaryHeapSuite {
+  @Test
+  def insertOneIsMax() {
+    val bh = new BinaryHeap[Int]()
+    bh.insert(1, 10)
+    assert(bh.max() === 1)
+    assert(bh.size === 1)
+    assert(bh.max() === 1)
+    assert(bh.size === 1)
+    assert(bh.contains(1) === true)
+    assert(bh.extractMax() === 1)
+    assert(bh.contains(1) === false)
+    assert(bh.size === 0)
+
+    intercept[Exception](bh.max())
+    intercept[Exception](bh.extractMax())
+  }
+
+  @Test
+  def twoElements() {
+    val bh = new BinaryHeap[Int]()
+    bh.insert(1, 5)
+    assert(bh.contains(1) == true)
+    bh.insert(2, 10)
+    assert(bh.contains(2) == true)
+    assert(bh.max() === 2)
+    assert(bh.size === 2)
+    assert(bh.max() === 2)
+    assert(bh.size === 2)
+    assert(bh.extractMax() === 2)
+    assert(bh.contains(2) == false)
+    assert(bh.size === 1)
```

Comment:
Why are you repeating the checks of the `size`? I assume you are repeating `max` to ensure it's not popping the max element.